### ISSUE TITLE
[Bug] ConstantDiagLinearOperator._mul_constant

### ISSUE THREAD

**Issue by j-wilson** [cornellius-gp/linear_operator#24]
# 🐛 Bug

ConstantDiagLinearOperator._mul_constant seems to be bugged.

## To reproduce

```python
from linear_operator import operators
A = operators.DenseLinearOperator(rand(4, 2, 3, 3))
B = A.add_jitter(1.0)
C = operators.ConstantMulLinearOperator(B, rand([]))
C[..., -1:, :]
```

##  Stack trace/error message
```
The size of tensor a (2) must match the size of tensor b (4) at non-singleton dimension 1
---------------------------------------------------------------------------
RuntimeError                              Traceback (most recent call last)
<ipython-input-11-690c97df3908> in <module>
----> 1 C[..., -1:, :]
/mnt/xarfuse/uid-22150/d091ed77-seed-nspid4026533386_cgpid5352405-ns-4026533383/linear_operator/operators/_linear_operator.py in __getitem__(self, index)
   2632             res = self._get_indices(row_index, col_index, *batch_indices)
   2633         else:
-> 2634             res = self._getitem(row_index, col_index, *batch_indices)
   2635 
   2636         # If we selected a single row and/or column (or did tensor indexing), we'll be retuning a tensor
/mnt/xarfuse/uid-22150/d091ed77-seed-nspid4026533386_cgpid5352405-ns-4026533383/linear_operator/operators/constant_mul_linear_operator.py in _getitem(self, row_index, col_index, *batch_indices)
    102         constant = self._constant.expand(self.batch_shape)[batch_indices]
    103         constant = constant.view(*constant.shape, 1, 1)
--> 104         return base_linear_op * constant
    105 
    106     def _matmul(self, rhs):
/mnt/xarfuse/uid-22150/d091ed77-seed-nspid4026533386_cgpid5352405-ns-4026533383/linear_operator/operators/_linear_operator.py in __mul__(self, other)
   2665     @_implements_second_arg(torch.Tensor.mul)
   2666     def __mul__(self, other: Union[torch.Tensor, LinearOperator, float]) -> LinearOperator:
-> 2667         return self.mul(other)
   2668 
   2669     @_implements_second_arg(torch.Tensor.add)
/mnt/xarfuse/uid-22150/d091ed77-seed-nspid4026533386_cgpid5352405-ns-4026533383/linear_operator/operators/_linear_operator.py in mul(self, other)
   1753                 return self._mul_constant(other.squeeze())
   1754             elif other.shape[-2:] == torch.Size((1, 1)):
-> 1755                 return self._mul_constant(other.view(*other.shape[:-2]))
   1756 
   1757         return self._mul_matrix(to_linear_operator(other))
/mnt/xarfuse/uid-22150/d091ed77-seed-nspid4026533386_cgpid5352405-ns-4026533383/linear_operator/operators/sum_linear_operator.py in _mul_constant(self, other)
     44     def _mul_constant(self, other):
     45         # We're using a custom method here - the constant mul is applied to the base_linear_ops
---> 46         return self.__class__(*[lt._mul_constant(other) for lt in self.linear_ops])
     47 
     48     def _bilinear_derivative(self, left_vecs, right_vecs):
/mnt/xarfuse/uid-22150/d091ed77-seed-nspid4026533386_cgpid5352405-ns-4026533383/linear_operator/operators/interpolated_linear_operator.py in _mul_constant(self, other)
    218         # This preserves the interpolated structure
    219         return self.__class__(
--> 220             self.base_linear_op._mul_constant(other),
    221             self.left_interp_indices,
    222             self.left_interp_values,
/mnt/xarfuse/uid-22150/d091ed77-seed-nspid4026533386_cgpid5352405-ns-4026533383/linear_operator/operators/diag_linear_operator.py in _mul_constant(self, constant)
    272 
    273     def _mul_constant(self, constant: Tensor) -> "ConstantDiagLinearOperator":
--> 274         return self.__class__(self.diag_values * constant, diag_shape=self.diag_shape)
    275 
    276     def _mul_matrix(self, other: Union[Tensor, LinearOperator]) -> Union[Tensor, LinearOperator]:
RuntimeError: The size of tensor a (2) must match the size of tensor b (4) at non-singleton dimension 1
```

## Additional context
The following seems to work, but I don't know enough to claim that it's a fix.
```
from unittest.mock import patch


def _mul_constant(self, constant):
    return self.__class__(self.diag_values * constant.unsqueeze(-1), diag_shape=self.diag_shape)


with patch.object(operators.ConstantDiagLinearOperator, "_mul_constant", new=_mul_constant):
    C[..., -1:, :]
  ```

**Comment by gpleiss** [comment]
That fix seems reasonable, if you want to submit a PR.

**Comment by JonathanWenger** [comment]
Fixed by #37.
